File: day1/day1.jl
# function stringNumParser(line::String)
#     numberDict:: Dict = Dict("one"=>"1","two"=>"2","three"=>"3","four"=>"4","five"=>"5","six"=>"6","seven"=>"7","eight"=>"8","nine"=>"9")
#     for key in keys(numberDict)
#         i = findfirst(key,line)
#         if i !== nothing
#             print(line[i])

#         end
#     end
# end

function stringNumParser(line::String)
    numberDict:: Vector{Pair{String,String}} = ["one"=>"1","two"=>"2","three"=>"3","four"=>"4","five"=>"5","six"=>"6","seven"=>"7","eight"=>"8","nine"=>"9"]
    firstI = nothing
    firstPair = nothing

    for replacer in numberDict
        i = findfirst(replacer[1],line)
        if i !== nothing
            if isnothing(firstI)
                firstI = i
                firstPair = replacer
            else    
                if i < firstI
                    firstI = i
                    firstPair = replacer
                end
            end
        end
    end
    lastI = nothing
    lastPair = nothing
    for replacer in numberDict
        i = findlast(replacer[1],line)
        if i !== nothing
            if isnothing(lastI)
                lastI = i
                lastPair = replacer
            else    
                if i > lastI
                    lastI = i
                    lastPair = replacer
                end
            end
        end
    end

    if !isnothing(firstPair)
        line = replace(line,firstPair)
    end
    if !isnothing(lastPair)
        line = replace(line,lastPair)
    end
    return line
end


function main()
    calNum :: Int64 = 0
    open("day1/input.txt","r") do f
        while ! eof(f)
            lineCalNum :: String = ""
            lineForward:: String = readline(f)
            
            # lineForward = stringNumParser(lineForward) #day2

            lineReverse:: String = reverse(lineForward)
            for charForward in lineForward
                if isnumeric(charForward)
                    lineCalNum *= charForward
                    break
                end
            end
            for charReverse in lineReverse
                if isnumeric(charReverse)
                    lineCalNum *= charReverse
                    break
                end
            end
            calNum += parse(Int,lineCalNum)
        end
    end
    println(calNum)
end

main()
# println(stringNumParser("eightthreewothree"))

File: day1/day1.py
import re

def findAndReplaceFirst(inputString:str,matches:dict):
    minIndex = 1000
    minMatch = None
    for match in matches:
        i = inputString.find(match)
        if i == -1:
            i = 1000
        if i < minIndex:
            minIndex = i
            minMatch = match
    if minMatch:
        return inputString.replace(minMatch,matches[minMatch],1)
    else:
        return inputString

mapDict = {"one":"1","two":"2","three":"3","four":"4","five":"5","six":"6","seven":"7","eight":"8","nine":"9"}
revMapDict = {k[::-1]:v for k,v in mapDict.items()}


def part2LineParser(line:str):
    line = findAndReplaceFirst(line,mapDict)
    line = findAndReplaceFirst(line[::-1],revMapDict)[::-1]
    return line
    




def main():
    telsum = 0
    with open("day1/input.txt") as file:
        content = file.readlines()
        for line in content:
            line = line.strip()
            line = part2LineParser(line)
            for charFor in line:
                if charFor.isdigit():
                    break
            for charRev in reversed(line):
                if charRev.isdigit():
                    break
            telsum += int(charFor+charRev)
        
    print(telsum)


if __name__ == '__main__':
    main()
    # print(part2LineParser("sixdddkcqjdnzzrgfourxjtwosevenhg9"))